### DISCUSSION TITLE
Detecting when scroll goes up or down?

### DISCUSSION THREAD

**Discussion by jzhang172** [idiotWu/smooth-scrollbar#526] [General]
I want some events to happen depending on if scrolling up or down such as a navbar appearing, how do I detect the scrolling/direction?

https://stackblitz.com/edit/angular-ivy-nwkcnv?file=src%2Fapp%2Fapp.component.ts,src%2Fapp%2Fapp.component.html

**Comment by sadeghbarati** [comment]
https://github.com/idiotWu/smooth-scrollbar/discussions/346

Might have better API in v9

**Reply by jzhang172** [reply]
> #346
> 
> Might have better API in v9

Thank you, works great...I am putting this inside my ngOnInit since I am using Angular, is this the correct placement or does it not matter where I put it?

**Reply by sadeghbarati** [reply]
You should place this code in ogOnInit (onMount) right after you initialize the scrollbar


```ts
ngOnInit() {
    let element = notBodyTag;

    const scrollbar = Scrollbar.init(element);
    
    scrollbar.addListener(({ offset }) => {
      //...
    });
}
```

**Reply by sadeghbarati** [reply]
If you using [GSAP ScrollTrigger](https://greensock.com/docs/v3/Plugins/ScrollTrigger/direction) + `smooth-scrollbar` you should take a look at their docs much easier than this way

Read more GSAP ScrollTrigger and smooth-scrollbar integration here
https://greensock.com/docs/v3/Plugins/ScrollTrigger/static.scrollerProxy()

**Reply by jzhang172** [reply]
> If you using [GSAP ScrollTrigger](https://greensock.com/docs/v3/Plugins/ScrollTrigger/direction) + `smooth-scrollbar` you should take a look at their docs much easier than this way
> 
> Read more GSAP ScrollTrigger and smooth-scrollbar integration here https://greensock.com/docs/v3/Plugins/ScrollTrigger/static.scrollerProxy()

Okay great!  Thank you!
